factory.rb

```diff
@@ -1,139 +1,138 @@
-class Factory 
+class Factory
 
 	include Enumerable
-	
-	 def self.new(*args, &block) 
-	 	
+
+	 def self.new(*args, &block)
+
 	 	class_name = args.shift  if (args[0].is_a?(String) && args[0].match(/[A-Z]/))
-         
-		obj = Class.new  do 	
 
-			attr_accessor :vars	
+		obj = Class.new  do
+
+			attr_accessor :vars
 
 			define_method :initialize do |*vars|
 				raise "ArgumentError" unless vars.count == args.count
 				@args = args
-				@vars = vars															
-			end	
-			
-		    class_eval(&block) if block
+				@vars = vars
+			end
+
+		  class_eval(&block) if block
```

Comment:
Remember about open class? this declaration won't be able to rewrite default behavior. Place this somewhere in the end of class definition block